src/analyses/base.ml

```diff
@@ -1257,7 +1257,7 @@ struct
             | _ -> None
           in
           let alen = List.filter_map (fun v -> lenOf v.vtype) (AD.to_var_may a) in
-          let d = List.fold_left ID.join (ID.bot_of (Cilfacade.ptrdiff_ikind ())) (List.map (ID.of_int (Cilfacade.ptrdiff_ikind ()) %BI.of_int) (slen @ alen)) in
+          let d = Seq.fold_left ID.join (ID.bot_of (Cilfacade.ptrdiff_ikind ())) ((Seq.map (ID.of_int (Cilfacade.ptrdiff_ikind ()) %BI.of_int))@@ List.to_seq (slen @ alen)) in
```

Comment:
Can one do better than `List.to_seq` after the append here?

Reply:
Also, I'd prefer a space before the `@@`.

Reply:
> Can one do better than `List.to_seq` after the append here?

Thanks for the feedback, I will try to refactor the `List.to_seq` here.